src/gateway/delegate-routing.ts

```diff
@@ -31,34 +38,73 @@ const onlines: { [index: string]: RoutingData } = {};
 router.use(
   "/findpeer",
   mixGetPost,
+  rateLimit({
+    enabled: gatewayConfigs.delegates.rateLimit.findPeerEnabled,
+    points: gatewayConfigs.delegates.rateLimit.findPeerLimit,
+    duration: gatewayConfigs.delegates.rateLimit.findPeerDuration,
+  }),
   asyncHandler(async (req, res, next) => {
     // @ts-ignore
-    let { id } = req.mixed;
+    const {id, timestamp, signature, requesterId} = req.mixed;
 
-    if (!id) throw `Missing parameter 'id'`;
+    if (!id || !timestamp || !signature || !requesterId)
+      throw `Missing parameters`;
 
-    const peerInfos: MuonNodeInfo[] = await NetworkIpc.filterNodes({
-      list: [id],
+    validateTimestamp(timestamp, findPeerValidPeriod);
+    const nodesInfo: MuonNodeInfo[] = await NetworkIpc.filterNodes({
```

Comment:
There is no need to load data from NodeManager. Everything should be on saved on `onlies`